lib/Jit/LLILCJit.cpp

```diff
@@ -560,69 +565,74 @@ void ObjectLoadListener::getDebugInfoForObject(
     if (!AddrOrError) {
       continue; // Error.
     }
-    uint64_t Addr = AddrOrError.get();
-    uint64_t Size = Pair.second;
+    uint64_t SingleAddr = AddrOrError.get();
+    uint64_t SingleSize = Pair.second;
+    if (SingleAddr < Addr) {
+      // The main function is always laid out first
+      Addr = SingleAddr;
+    }
+    Size += SingleSize;
+  }
 
-    uint32_t LastDebugOffset = (uint32_t)-1;
-    uint32_t NumDebugRanges = 0;
-    ICorDebugInfo::OffsetMapping *OM;
+  uint32_t LastDebugOffset = (uint32_t)-1;
```

Comment:
I don't know why the diff isn't figuring this out, but fyi everything from here down is just an indentation change (and removing the closing brace where there used to be a loop).